test/dialects/dialects_test.py

```diff
@@ -48,6 +48,15 @@ def test__dialect__base_file_parse(dialect, file):
         assert "unparsable" not in typs
 
 
+@pytest.mark.parametrize("dialect,file", parse_success_examples)
+def test__dialect__base_broad_fix(dialect, file, raise_critical_errors_after_fix):
+    """For given test examples, check successful parsing."""
+    raw = load_file(dialect, file)
+    # Load the right dialect
+    config = FluffConfig(overrides=dict(dialect=dialect))
+    Linter(config=config).lint_string_wrapped(raw, fix=True)
+
+
```

Comment:
@barrywhart  I think this is an improvement over the bash script you added recently (also edited below).

It does basically the same thing but can actually be debugged in python. It could probably also be sped up with caching I guess.

As mentioned in the function above from which I derived it, there is way too much logging - can this be suppressed? (its not needed for the parsing for example)

Reply:
Can you tell me more about the logging? Python logging supports different levels. When you run the tests, logging and printing are suppressed/captured by default.

Reply:
BTW, @tunetheweb wrote the bash script, not me.

Reply:
Fair enough, I guess I was tired by the time I was writing this and Id just seen your name under so many of the blame lines :)

Pytest suppresses logging but it captures it all and spits it out when a test fails - (only the output related to that test). So using this over the bash means we --fix all the files but actually reject and raise and therefore fail and get output about the SQL + rule causing the "Critical error"

Reply:
I’m totally cool with changing this from bash to Python. We talked about it at the time as would also allow Wndows to run it.

It was as a quick hack to make use of our many fixtures files (which were previously only used for parse testing) to also test rules - initially`lint` and then, as of #2830 merged last might, also `fix`. 

However, from a quick look on my phone, it looks like you’ve only implemented the `lint` version? As I say #2830 includes `fix`. Just need to be careful as that changes the file, so need to revert the change at the end. And also ensure no other test is running at the same time.

I also think this is a big enough change so it should be a separate PR.